src/target/ext/offsets.rs

```diff
@@ -0,0 +1,18 @@
+//! Handle the `qOffsets` command
+
+use crate::target::Target;
+
+/// Target Extension - Handle the `qOffsets` command
+pub trait OffsetsCmd: Target {
+    /// Handle the `qOffsets` command
+    ///
+    /// For some targets, sections may be relocated from their base address. As
+    /// a result, the stub may need to tell GDB the final section addresses
+    /// to ensure that debug symbols are resolved correctly after relocation.
+    ///
+    /// Implementing this command allows the stub to report text, data, and bss
+    /// offsets to GDB.
```

Comment:
Could you add a `/// _Note:_` which mentions that this feature uses the legacy `qOffsets` packet, and that it would be good to add support for the newer and more flexible `qXfer:libraries:read` packet instead (linking back to #20)? The new packet support relocating arbitrary sections (not just Text/Data/BSS).